src/sql/src/parsers/alter_parser.rs

```diff
@@ -51,9 +51,20 @@ impl<'a> ParserContext<'a> {
                     parser.peek_token()
                 )));
             }
+        } else if parser.parse_keyword(Keyword::RENAME) {
+            let new_table_name_obj = parser.parse_object_name()?;
+            let new_table_name: String = match &new_table_name_obj.0[..] {
+                [table] => Ok(table.value.clone()),
+                [_schema, table] => Ok(table.value.clone()),
+                _ => Err(ParserError::ParserError(format!(
+                    "expect table name to be <schema>.<table> or <table>, actual: {new_table_name_obj}"
+                ))),
+            }
+            .unwrap();
```

Comment:
Several points:
- We don't support renaming tables across schema, so we should return an error instead of silently ignoring the schema in the object name, which might confuse users.
- The unwrap is unnecessary
```suggestion
            let new_table_name = match &new_table_name_obj.0[..] {
                [table] => table.value.clone(),
                _ => return Err(ParserError::ParserError(format!(
                    "expect table name, actual: {new_table_name_obj}"
                ))),
            };
```